File: entities/src/main/java/com/jbrunton/entities/Configuration.java
package com.jbrunton.entities;

import com.google.auto.value.AutoValue;
import com.google.common.base.Optional;

@AutoValue
public abstract class Configuration {
    public abstract String secureBaseUrl();

    public static Configuration create(String secureBaseUrl) {
        return new AutoValue_Configuration(secureBaseUrl);
    }

    public Optional<String> expandUrl(String relativePath) {
        if (relativePath == null) {
            return Optional.absent();
        }
        String expandedUrl = secureBaseUrl() + "w300" + relativePath;
        return Optional.of(expandedUrl);
    }
}

File: entities/src/main/java/com/jbrunton/entities/Configuration.kt
package com.jbrunton.entities

import com.google.common.base.Optional

data class Configuration(val secureBaseUrl: String) {
    fun expandUrl(relativePath: String?): Optional<String> {
        if (relativePath == null) {
            return Optional.absent()
        }
        val expandedUrl = secureBaseUrl + "w300" + relativePath
        return Optional.of(expandedUrl)
    }
}
